### PR TITLE
Fix emoji only filter

### PR DESCRIPTION
**Pull request by mario**
Previously, when you enabled emoji-only filter and focused the edittext the emoji popup would appear.

In the future, when you tried focusing it for the second time only a keyboard would open as in reality edittext is already focused. 

We now clear the focus after dismissing the emoji popup in case keyboard input is disabled.

### CODE REVIEW

**Comment by mario** [comment]
@vanniktech mind fixing the warnings here?

**Comment by vanniktech** [comment]
@mario I can do that once you're rebased

**Comment by mario** [comment]
@rubengees fancy fixing warnings? My head is a mess today.

**Comment by rubengees** [comment]
@mario Sure, done.

**Comment by rubengees** [comment]
I found a way to break out of the emoji picker and show a normal keyboard.

Steps:

- Open keyboard.
- Open input method dialog (for me on Android 10 in the bottom right).
- Select other keyboard.

**Comment by rubengees** [comment]
PMD reports the `EmojiPopup` as a possible god class now due to it's size, that might be harder to fix...

**Comment by vanniktech** [comment]
Ignore it

File: emoji/build.gradle [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies {
   testImplementation rootProject.ext.testing.assertJ
   testImplementation rootProject.ext.testing.robolectric
   testImplementation rootProject.ext.testing.privateConstructor
+  androidTestAnnotationProcessor deps.support.test.autoService
```
Comment (vanniktech) [inline emoji/build.gradle]: What do we need this for?

File: app/build.gradle [hunk 2]
```diff
@@ -68,6 +68,7 @@ dependencies {
   androidTestImplementation deps.support.test.runner
   androidTestImplementation deps.support.test.rules
   androidTestImplementation deps.fastLaneScreenGrab
+  androidTestAnnotationProcessor deps.support.test.autoService
```
Comment (vanniktech) [inline app/build.gradle]: same

File: emoji/build.gradle (hunk 1 above)
Comment (mario) [reply emoji/build.gradle]: One of the lints complained about it missing.

**Comment by mario** [comment]
> I found a way to break out of the emoji picker and show a normal keyboard.
> 
> Steps:
> 
> * Open keyboard.
> * Open input method dialog (for me on Android 10 in the bottom right).
> * Select other keyboard.

@vanniktech did you test this? Since I don't even see the input method dialog?

**Comment by vanniktech** [comment]
> did you test this? Since I don't even see the input method dialog?

Not yet

**Comment by vanniktech** [comment]
I found one way:

1. Check the checkbox
2. Click on the input field
3. Click on the dialog button
4. Dismiss the dialog
5. Click on the input field (Normal keyboard is shown instead of the emoji one)

**Comment by mario** [comment]
Should be ok now @vanniktech and @rubengees

**Comment by vanniktech** [comment]
@rubengees @mario how should we proceed with PR's so that they don't stay open that long?

I propose:

- At least one review
- Waiting for a second review for up to 3 days

**Comment by mario** [comment]
Well in general I'm highly in favour of at least two reviews, but I guess we need a policy on how long we wait. I feel like 3 days is too little? How does 7 days work for you two?

**Comment by vanniktech** [comment]
Sure. I'm fine with 7 days!

**Comment by vanniktech** [comment]
I'll merge this now as discussed with @mario.

**Comment by rubengees** [comment]
Sorry for being away so long, I agree regarding the proposed time for waiting.

**Comment by vanniktech** [comment]
No need to be sorry. I hope everything is fine @ruben

**Comment by rubengees** [comment]
Yes sure, was just a bit lazy in the last time 😄

**Pull request merged**